eth/catalyst/api.go

```diff
@@ -469,10 +469,16 @@ func (api *ConsensusAPI) checkInvalidAncestor(check common.Hash, head common.Has
 		}
 		api.invalidTipsets[head] = invalid
 	}
+	// if the last valid hash is the terminal pow block, return 0x0 for latest valid hash
+	lastValid := &invalid.ParentHash
+	if header := api.eth.BlockChain().GetHeaderByHash(invalid.ParentHash); header != nil && header.Difficulty.BitLen() == 0 {
```

Comment:
BTW, a better way of checking for zero is `integer.Sign() == 0`.

Reply:
Fixed